src/net.h

```diff
@@ -421,19 +422,19 @@ struct CombinerAll
     }
 };
 
-// Signals for message handling
-struct CNodeSignals
+// Interface for message handling
+class NetEventsInterface
 {
-    boost::signals2::signal<bool (CNode*, CConnman&, std::atomic<bool>&), CombinerAll> ProcessMessages;
-    boost::signals2::signal<bool (CNode*, CConnman&, std::atomic<bool>&), CombinerAll> SendMessages;
-    boost::signals2::signal<void (CNode*, CConnman&)> InitializeNode;
-    boost::signals2::signal<void (NodeId, bool&)> FinalizeNode;
+protected:
+    NetEventsInterface() = default;
+public:
+    virtual ~NetEventsInterface() = default;
```

Comment:
Well my objection was just that it forces you into a virtual destructor unless you remember to call derived classes final, and since I dont think anything should ever have anything other than a reference to (ie will not destruct directly) a NetEventsInterface, no need to make anything virtual. I don't care too strongly, however.